pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -106,6 +108,9 @@
   private static final RetryPolicy DEFAULT_RETRY_POLICY = RetryPolicies.exponentialBackoffRetryPolicy(5, 1000L, 2.0f);
 
   private final Map<String, SegmentAssignmentStrategy> _segmentAssignmentStrategyMap = new HashMap<>();
+  private final Map<String, Map<String, Long>> _segmentCrcMap = new ConcurrentHashMap<>();
```

Comment:
I don't think ConcurrentHashMap would help to keep it correct. Let's make getSegmentsCrcForTable synchronized, and change these two maps to normal HashMap?